docs/adr/ADR-001-Theme_and_tokens.md

```diff
@@ -0,0 +1,111 @@
+# ADR-001: Theme and tokens
+
+## Date
+
+2023-10-04
+
+## Status
+
+PROPOSED
+
+## Context
+
+At the moment, when providing custom themes to the Design System, the user can only override the palette of colors.
+In some cases, this can be limiting, as the user may need to override specific parts of components to respect A11Y
+guidelines or to have more coherent colors.
+
+## Decision
+
+We will change the structure of the Theme object provided to the Design System to allow providing overrides for either
+the palette, aliases or components tokens.
+Any change in the palette will be reflected in the components tokens.
+Any change in the aliases tokens will be reflected in the components tokens.
+Components tokens will accept either color tokens from the palette or aliases tokens.
+At the moment, only colors can be overridden, but this could be extended to other tokens in the future.
+
+The input structure accepted by the `DesignSystem` provider would be as follows:
+
+```typescript
+interface Palette {
+    color-brand-<tint>: string; // e.g: color-brand-05: #E0F0F9
+    color-accent-<tint>: string;
+    color-neutral-<tint>: string;
+    color-alert-<tint>: string;
+    color-informative-<tint>: string;
+    color-success-<tint>: string;
+    color-warning-<tint>: string;
+    ...
+}
+
+type RefTokensCustomization = Palette; // For now, only Palette would be customizable
+
+interface AliasesTokens {
+    // This to be determined, but would be something like
+    button-color-secondary: keyof AliasesTokens | keyof RefTokens;
+    interaction-color: keyof AliasesTokens | keyof RefTokens;
+};
+
+interface ComponentsTokens {
+    <component>-<flavor?>-<property>: keyof AliasesTokens | keyof RefTokens;
+    button-primary-background-color: 'color-brand-05';
+    ...
+};
+
+interface ThemeCustomization {
+    ref: RefTokensCustomization;
+    aliases: AliasesTokens; // We might need a AliasesTokensCustomization layer to prevent overriding certain aliases 
```

Comment:
Minor details, but I feel like the the first word should not be plural when describing token types (i.e. 'component tokens' instead of 'components tokens', AliasTokens instead of AliasesToken).

However, I don't mind using plural for the keys.